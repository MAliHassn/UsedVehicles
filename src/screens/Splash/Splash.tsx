import { Image, StyleSheet, Text, View } from 'react-native';
import React from 'react';
import { Colors } from '../../constants/Colors';
import { images } from '../../constants/Images';
import { SCREEN_HEIGHT, SCREEN_WIDTH } from '../../constants/Screen';
import { Fonts } from '../../constants/Fonts';
import {useSplash} from './Hooks';

const Splash = () => {
  const {} = useSplash();

  return (
    <View style={styles.container}>
      <Image source={images.app_logo} style={styles.image} resizeMethod="auto" />

      <View style={styles.bottom_view}>
        <Text style={styles.text}>Powered by</Text>
        <Image source={images.rak_bank} style={styles.rak_bank_image} resizeMethod="auto" />
      </View>

      <Image
        source={images.splashBg}
        style={styles.bottomBg}
        resizeMode="stretch"
      />

    </View>
  );
};

export default Splash;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.white
  },
  image: {
    aspectRatio: 2,
    height: "auto",
    width: SCREEN_WIDTH * 0.7,
  },
  bottom_view: {
    position: "absolute",
    bottom: 10,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center"
  },
  text: {
    color: Colors.black,
    fontSize: Fonts.body
  },
  rak_bank_image: {
    aspectRatio: 1,
    height: "auto",
    width: SCREEN_WIDTH * 0.15,
  },
  bottomBg: {
    position: 'absolute',
    bottom: 30,
    width: SCREEN_WIDTH,
    height:SCREEN_HEIGHT/2.5,
    zIndex:-1
  },
});