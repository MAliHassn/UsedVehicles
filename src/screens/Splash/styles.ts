import {StyleSheet} from 'react-native';
import {useCustomTheme} from '../../constants/useCustomTheme';
import {SCREEN_HEIGHT, SCREEN_WIDTH} from '../../constants/Screen';

const useStyles = () => {
  const {colors, fonts} = useCustomTheme();
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: colors.white,
    },
    image: {
      aspectRatio: 2,
      height: 'auto',
      width: SCREEN_WIDTH * 0.7,
    },
    bottom_view: {
      position: 'absolute',
      bottom: 10,
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
    },
    text: {
      color: colors.black,
      fontSize: fonts.body,
    },
    rak_bank_image: {
      aspectRatio: 1,
      height: 'auto',
      width: SCREEN_WIDTH * 0.15,
    },
    bottomBg: {
      position: 'absolute',
      bottom: 30,
      width: SCREEN_WIDTH,
      height: SCREEN_HEIGHT / 2.5,
      zIndex: -1,
    },
  });

  return {
    styles,
  };
};

export default useStyles;
