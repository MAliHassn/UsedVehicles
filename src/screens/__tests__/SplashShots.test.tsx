import React from 'react';
import renderer from 'react-test-renderer';
import {NavigationContainer} from '@react-navigation/native';
import {Splash} from '../Splash';
import {waitFor} from '@testing-library/react-native';

it('renders correctly', async () => {
  const component = (
    <NavigationContainer>
      <Splash />
    </NavigationContainer>
  );
  const rendered = renderer.create(component).toJSON();
  await waitFor(() => expect(rendered).toMatchSnapshot());
});
