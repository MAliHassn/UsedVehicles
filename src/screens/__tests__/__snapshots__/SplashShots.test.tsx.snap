// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMethod="auto"
    source={
      {
        "testUri": "../../../src/assets/ui/applogo.png",
      }
    }
    style={
      {
        "aspectRatio": 2,
        "height": "auto",
        "width": 525,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "bottom": 10,
        "flexDirection": "row",
        "justifyContent": "center",
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 12,
        }
      }
    >
      Powered by
    </Text>
    <Image
      resizeMethod="auto"
      source={
        {
          "testUri": "../../../src/assets/ui/rakbank-logo-news.png",
        }
      }
      style={
        {
          "aspectRatio": 1,
          "height": "auto",
          "width": 112.5,
        }
      }
    />
  </View>
  <Image
    resizeMode="stretch"
    source={
      {
        "testUri": "../../../src/assets/ui/splash_bg.png",
      }
    }
    style={
      {
        "bottom": 30,
        "height": 533.6,
        "position": "absolute",
        "width": 750,
        "zIndex": -1,
      }
    }
  />
</View>
`;
